lib/models/graph-object.js

```diff
@@ -56,12 +56,6 @@ function GraphModelFactory (
                 var obj = this.toObject();
                 return Constants.Task.ActiveStates.indexOf(obj._status) > -1;
             }
-        },
-        $indexes: [
-            {
-                keys: { instanceId: 1 },
```

Comment:
previously, every index is carefully designed to satisfy the frequently database query, so if index is removed, the performance will certainly be impacted. Usually, you could only see the performance issue during scale environment. So I'd suggest you doing some scale testing before removing these indexes.

From the implementation of waterline mongo adapter, see if the "unique" of one attribute is set as true, then waterline will automatically create an unique index for it (see code https://github.com/balderdashy/sails-mongo/blob/7572dd47a1e4b523d728e6ad4cd30bdf6b4174fc/lib/index.js#L768). So this may be the source of duplicated index, we could remove those index setting if the attribute is set as "true", or we can always remove the unique from attribute and keep using our own index setting.